test/unit/utils/deep_copy.js

```diff
@@ -129,26 +129,28 @@ describe("utils.deep_copy", () => {
 	});
 
 	it("Should return a deep copy of the passed DynamoDB set", () => {
-		const original = dynamoose.aws.converter().output({"SS": ["Hello", "World", "Universe"]});
+		const original = dynamoose.aws.converter().convertToNative({"SS": ["Hello", "World", "Universe"]});
 		const copy = utils.deep_copy(original);
-		expect(copy).to.deep.equal(dynamoose.aws.converter().output({"SS": ["Hello", "World", "Universe"]}));
 
-		original.values[0] = "Welcome";
+		expect(copy).to.deep.equal(dynamoose.aws.converter().convertToNative({"SS": ["Hello", "World", "Universe"]}));
 
-		expect(original).to.deep.equal(dynamoose.aws.converter().output({"SS": ["Welcome", "World", "Universe"]}));
-		expect(copy).to.deep.equal(dynamoose.aws.converter().output({"SS": ["Hello", "World", "Universe"]}));
+		original.delete("Universe");
+		original.add("Peace");
+
+		expect(original).to.deep.equal(dynamoose.aws.converter().convertToNative({"SS": ["Hello", "World", "Peace"]}));
 		expect(copy.constructor).to.deep.equal(original.constructor);
 	});
```

Comment:
These 2 tests should fail if we change:

```
if (obj.constructor !== Object) {
    copy = Object.assign(Object.create(Object.getPrototypeOf(obj)), obj);
} else {
    copy = {};
}
```

To:

```
copy = {};
```

In `deep_copy.ts`.

As they do on v2. However on this PR they don't fail with that code change.

Reply:
Same with the `Should throw when passed an invalid DynamoDB set` test below

Reply:
As discussed in Slack, this is actually fine, since AWS SDK v3 changed how this works.

But we should add the following test:

```
it("Should return a deep copy of the passed multiple nested class instances when errors are thrown on constructor for undefined arguments", () => {
	class NameWrapper {
		constructor (name) {
			if (!name) {
				throw new Error("Name is required");
			}
			this.name = name;
		}
	}

	class PersonWrapper {
		constructor (name, age) {
			if (!name) {
				throw new Error("Name is required");
			}
			if (!age) {
				throw new Error("Age is required");
			}
			this.name = name;
			this.age = age;
		}
	}
	const original = new PersonWrapper(new NameWrapper("Tim"), 20);
	const copy = utils.deep_copy(original);
	expect(copy).to.deep.equal(new PersonWrapper(new NameWrapper("Tim"), 20));

	original.name.name = "Bob";
	original.age = 25;

	expect(original).to.deep.equal(new PersonWrapper(new NameWrapper("Bob"), 25));
	expect(copy).to.deep.equal(new PersonWrapper(new NameWrapper("Tim"), 20));
	expect(original.constructor).to.deep.equal(PersonWrapper);
	expect(copy.constructor).to.deep.equal(PersonWrapper);
	expect(original.name.constructor).to.deep.equal(NameWrapper);
	expect(copy.name.constructor).to.deep.equal(NameWrapper);
});
```